crates/runtime/src/flight/middleware.rs

```diff
@@ -160,41 +179,45 @@ where
         self.inner.poll_ready(cx)
     }
 
-    fn call(&mut self, req: http::Request<ReqBody>) -> Self::Future {
+    fn call(&mut self, mut req: http::Request<ReqBody>) -> Self::Future {
         // Apply rate limiting to the Flight DoPut only
         if req.uri().path() != "/arrow.flight.protocol.FlightService/DoPut" {
             return Box::pin(self.inner.call(req));
         }
 
-        if let Err(wait_time) = self.rate_limiter.check() {
-            let retry_after_secs = wait_time
-                .wait_time_from(wait_time.earliest_possible())
-                .as_secs();
+        let rate_limiter = Arc::clone(&self.rate_limiter);
 
-            tracing::trace!("Request rate-limited, must retry after {retry_after_secs} seconds.",);
+        // Create a rate limit check that the downstream service can use to check if the request should be rate limited.
+        // We don't directly check the rate limit here because not all DoPut requests should be rate limited
+        // (i.e. binding parameters to parameterized queries).
+        let check: Arc<RateLimitCheckFn> = Arc::new(move || {
```

Comment:
Does this checker get applied anywhere except doPut? How/where does it do that?

Reply:
It's currently only for DoPut, that is checked by the URI above